internal/accrual/accrual.go

```diff
@@ -0,0 +1,93 @@
+package accrual
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"gophermart/internal/logger"
+	"gophermart/internal/models"
+	"gophermart/internal/store"
+	"io"
+	"net/http"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+const urlGetUserOrders = "%s/api/orders/%d" // получение информации о расчёте начислений баллов лояльности
+
+var ErrStatusNoContent = errors.New("StatusNoContent")
+var ErrStatusTooManyRequests = errors.New("StatusTooManyRequests")
+var ErrStatusInternalServerError = errors.New("StatusInternalServerError")
+
+func PrepareBatch(storage *store.StorageContext) (statusOrders []int64) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	statusOrders, err := storage.GetOrdersProcessing(ctx)
+	if err != nil {
+		logger.Logger.Warn("Ошибка получения данных о заказах")
+		return statusOrders
+	}
+	logger.Logger.Info(fmt.Sprintf("%d", statusOrders))
+	return statusOrders
+}
+
+func UpdateStatusOrdersWorker(workerID int, storage *store.StorageContext, urlAccrual string, jobs <-chan int64) {
+	ctx := context.Background()
+	for job := range jobs {
+		logger.Logger.Info(fmt.Sprintf("Воркер %d", workerID))
+
+		statusOrder := GetStatus(ctx, job, urlAccrual)
+		if statusOrder != nil {
+			err := storage.UpdateStatusOrders(ctx, statusOrder)
+			if err != nil {
+				logger.Logger.Warn("Ошибка обновления данных", zap.Error(err))
+			}
+		}
+
+	}
+}
+
+func GetStatus(ctx context.Context, number int64, urlAccrual string) *models.StatusOrdersAccrual {
+	ctx, cansel := context.WithTimeout(ctx, time.Duration(30*time.Second))
```

Comment:
```suggestion
	ctx, cansel := context.WithTimeout(ctx, 30*time.Second)
```
лишнее преобразование